src/server/session.ts

```diff
@@ -541,31 +556,29 @@ module ts.server {
                             var editorOptions: ts.EditorOptions = {
                                 IndentSize: formatOptions.IndentSize,
                                 TabSize: formatOptions.TabSize,
-                                NewLineCharacter: "\n",
+                                NewLineCharacter: formatOptions.NewLineCharacter,
                                 ConvertTabsToSpaces: formatOptions.ConvertTabsToSpaces,
                             };
-                            var indentPosition =
-                                compilerService.languageService.getIndentationAtPosition(file, position, editorOptions);
+                            var preferredIndent = compilerService.languageService.getIndentationAtPosition(file, position, editorOptions);

```

Comment:
We should already have code that determines first non-whitespace column\character in the line, check [`SmartIndenter.findFirstNonWhitespaceCharacterAndColumn`](https://github.com/Microsoft/TypeScript/blob/master/src/services/formatting/smartIndenter.ts#L324)